test/controllers/declaration/TransportDetailsControllerSpec.scala

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.declaration
+
+import base.CSRFUtil._
+import base.CustomExportsBaseSpec
+import forms.Choice
+import forms.Choice.choiceId
+import forms.declaration.TransportDetails
+import generators.Generators
+import org.mockito.ArgumentMatchers
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito.{reset, verify}
+import org.scalacheck.Arbitrary._
+import org.scalatest.prop.PropertyChecks
+import play.api.data.Form
+import play.api.test.FakeRequest
+import play.api.test.Helpers._
+import play.twirl.api.Html
+import services.Countries
+import uk.gov.hmrc.auth.core.InsufficientEnrolments
+import views.html.declaration.transport_details
+
+class TransportDetailsControllerSpec extends CustomExportsBaseSpec with Generators with PropertyChecks {
+
+  private val uri = uriWithContextPath("/declaration/transport-details")
+
+  val form: Form[TransportDetails] = Form(TransportDetails.formMapping)
+
+  def view(form: Form[TransportDetails] = form, request: FakeRequest[_]): Html =
```

Comment:
The same thing like before about arguments order

Reply:
updated